src/main/kotlin/no/nav/familie/ks/sak/kjerne/brev/BrevUtil.kt

```diff
@@ -88,3 +93,99 @@ fun hentVedtaksbrevtype(
+    opplysningspliktHjemlerSkalMedIBrev: Boolean = false,
+    målform: Målform
+): String {
+    val sanityStandardbegrunnelser = brevVedtaksperioder.flatMap { vedtaksperiode ->
+        vedtaksperiode.begrunnelseMedDataFraSanity.mapNotNull { begrunnelse ->
+            begrunnelse.standardbegrunnelse.tilSanityBegrunnelse(sanityBegrunnelser)
+        }
+    }
+
+    val ordinæreHjemler =
+        hentOrdinæreHjemler(
+            hjemler = sanityStandardbegrunnelser.flatMap { it.hjemler }
+                .toMutableSet(),
+            opplysningspliktHjemlerSkalMedIBrev = opplysningspliktHjemlerSkalMedIBrev,
+            finnesVedtaksperiodeMedFritekst = brevVedtaksperioder.flatMap { it.fritekster }.isNotEmpty()
+        )
+
+    val alleHjemlerForBegrunnelser = hentAlleTyperHjemler(
+        ordinæreHjemler = ordinæreHjemler.distinct(),
+        målform = målform
+    )
+
+    return slåSammenHjemlerAvUlikeTyper(alleHjemlerForBegrunnelser)
+}
+
+private fun slåSammenHjemlerAvUlikeTyper(hjemler: List<String>) = when (hjemler.size) {
+    0 -> throw FunksjonellFeil("Ingen hjemler var knyttet til begrunnelsen(e) som er valgt. Du må velge minst én begrunnelse som er knyttet til en hjemmel.")
+    1 -> hjemler.single()
+    else -> slåSammenListeMedHjemler(hjemler)
+}
+
+private fun slåSammenListeMedHjemler(hjemler: List<String>): String {
+    return hjemler.reduceIndexed { index, acc, s ->
+        when (index) {
+            0 -> acc + s
+            hjemler.size - 1 -> "$acc og $s"
+            else -> "$acc, $s"
+        }
+    }
+}
+
+private fun hentAlleTyperHjemler(
+    ordinæreHjemler: List<String>,
+    målform: Målform
+): List<String> {
+    val alleHjemlerForBegrunnelser = mutableListOf<String>()
+
+    // Rekkefølgen her er viktig
+    if (ordinæreHjemler.isNotEmpty()) {
+        alleHjemlerForBegrunnelser.add(
+            "${
+            when (målform) {
+                Målform.NB -> "barnetrygdloven" // TODO hvilken lov? kontantstøtteloven?
+                Målform.NN -> "barnetrygdlova"
+            }
+            } ${
+            hjemlerTilHjemmeltekst(
+                hjemler = ordinæreHjemler,
+                lovForHjemmel = "barnetrygdloven"
+            )
+            }"
+        )
+    }
+    return alleHjemlerForBegrunnelser
+}
+
+fun hjemlerTilHjemmeltekst(hjemler: List<String>, lovForHjemmel: String): String {
+    return when (hjemler.size) {
+        0 -> throw Feil("Kan ikke lage hjemmeltekst for $lovForHjemmel når ingen begrunnelser har hjemler fra $lovForHjemmel knyttet til seg.")
+        1 -> "§ ${hjemler[0]}"
+        else -> "§§ ${slåSammen(hjemler)}"
+    }
+}
+
+private fun hentOrdinæreHjemler(
+    hjemler: MutableSet<String>,
+    opplysningspliktHjemlerSkalMedIBrev: Boolean,
+    finnesVedtaksperiodeMedFritekst: Boolean
+): List<String> {
+    if (opplysningspliktHjemlerSkalMedIBrev) {
+        val hjemlerNårOpplysningspliktIkkeOppfylt = listOf("17", "18")
```

Comment:
Tror det kun skal være ```"16"``` her for kontantstøtte.
https://lovdata.no/lov/1998-06-26-41/§16